### PR TITLE
error hierarchy

### PR DESCRIPTION
**Pull request by pshemass**

### CODE REVIEW

File: keeper/src/main/scala/zio/keeper/Error.scala [hunk 1]
```diff
@@ -3,36 +3,45 @@ package zio.keeper
...
-object Error {
+sealed abstract class Error(val msg: String = "")
 
-  case class ConnectionTimeout(addr: SocketAddress, timeout: Duration) extends Error
-  case class RequestTimeout(addr: SocketAddress)                       extends Error
+sealed abstract class AbstractSerializationError(msg: String = "") extends Error(msg = msg)
 
-  case class CannotFindSerializerForMessage[A](obj: A)    extends Error
-  case class CannotFindSerializerForMessageId(msgId: Int) extends Error
+case class SerializationTypeError[A](val cause: scala.Throwable)(implicit ct: ClassTag[A])
```
[the remark this answers is not in the document]
Comment (pshemass) [reply keeper/src/main/scala/zio/keeper/Error.scala]: I haven't removed ```ClassTag``` yet but I dropped scoping

File: keeper/src/main/scala/zio/keeper/Error.scala [hunk 2]
```diff
@@ -3,36 +3,45 @@ package zio.keeper
...
-//sealed trait Error extends Exception
-sealed abstract class Error(msg: String = "", cause: Throwable = null) extends Exception(msg, cause)
+import scala.reflect.ClassTag
 
-object Error {
+sealed abstract class Error(val msg: String = "")
 
-  case class ConnectionTimeout(addr: SocketAddress, timeout: Duration) extends Error
-  case class RequestTimeout(addr: SocketAddress)                       extends Error
+sealed abstract class AbstractSerializationError(msg: String = "") extends Error(msg = msg)
```
[the remark this answers is not in the document]
Comment (pshemass) [reply keeper/src/main/scala/zio/keeper/Error.scala]: done

File: keeper/src/main/scala/zio/keeper/errors.scala [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+
+import scala.reflect.ClassTag
+
+sealed abstract class Error(val msg: String = "")
+
+sealed abstract class SerializationError(msg: String = "") extends Error(msg = msg)
+
+object SerializationError {
+
+  final case class SerializationTypeError[A](val cause: Throwable)(implicit ct: ClassTag[A])
```
Comment (mschuwalow) [inline keeper/src/main/scala/zio/keeper/errors.scala]: I'm not super happy with us using class tags as they will bleed through everywhere we use this in a generic context

Comment (pshemass) [reply keeper/src/main/scala/zio/keeper/errors.scala]: I'm not happy either. Let me know how we can improve that.

**Pull request merged**